addons/l10n_tr_nilvera_einvoice/models/account_move.py

```diff
@@ -0,0 +1,253 @@
+        return super()._get_ubl_cii_builder_from_xml_tree(tree)
+
+    def _l10n_tr_nilvera_submit_einvoice(self, xml_file, customer_alias):
+        self._l10n_tr_nilvera_submit_document(
+            xml_file=xml_file,
+            endpoint=f"/einvoice/Send/Xml?Alias={customer_alias}",
+            success_message=_("The invoice has been successfully sent to Nilvera as an E-Invoice."),
+            post_series_func=self._l10n_tr_nilvera_post_einvoice_series,
+        )
+
+    def _l10n_tr_nilvera_submit_earchive(self, xml_file):
+        self._l10n_tr_nilvera_submit_document(
+            xml_file=xml_file,
+            endpoint="/earchive/Send/Xml",
+            success_message=_("The invoice has been successfully sent to Nilvera as an E-Arhive."),
+            post_series_func=self._l10n_tr_nilvera_post_earchive_series,
+        )
+
+    def _l10n_tr_nilvera_submit_document(self, xml_file, endpoint, success_message, post_series_func=None):
+        """Generic function to handle e-invoice and e-archive submissions to Nilvera."""
+        client = self.env.company._get_nilvera_client()
+        response = client.request(
+            "POST",
+            endpoint,
+            files={'file': (xml_file.name, xml_file, 'application/xml')},
+            handle_response=False,
+        )
+
+        msg = ""
+
+        if response.status_code == 200:
+            msg = success_message
+            self.l10n_tr_nilvera_send_status = 'sent'
+        elif response.status_code == 401:
+            raise UserError(_("Oops, seems like you're unauthorised to do this. Try another API key with more "
+                              "rights or contact Nilvera."))
+        elif 400 <= response.status_code < 500:
+            error_message, error_codes = self._l10n_tr_nilvera_einvoice_get_error_messages_from_response(response)
+
+            # If the sequence/series is not found on Nilvera, add it then retry.
+            if 3009 in error_codes and post_series_func:
+                post_series_func()
+                # post_series_func is not passed this time to avoid endless retries.
+                return self._l10n_tr_nilvera_submit_document(xml_file, endpoint, success_message)
+
+            msg += error_message
+
+        self.message_post(body=Markup(msg))
+
+    def _l10n_tr_nilvera_post_einvoice_series(self, client):
+        self._l10n_tr_nilvera_post_series("/einvoice/Series", client)
+
+    def _l10n_tr_nilvera_post_earchive_series(self, client):
+        self._l10n_tr_nilvera_post_series("/earchive/Series", client)
+
+    def _l10n_tr_nilvera_post_series(self, endpoint, client):
+        series = self.sequence_prefix.split('/')[0]
+        client.request(
+            "POST",
+            endpoint,
+            json={
+                'Name': series,
+                'IsActive': True,
+                'IsDefault': False,
+            },
+        )
+
+    def _l10n_tr_nilvera_get_submitted_document_status(self, client):
```

Comment:
Does this method need to take `client` as parameter? It is called twice and both times the client is created just before and then never used after. Wouldn't it possible to create the client from within this method?